x-pack/plugins/infra/server/routes/inventory_views/delete_inventory_view.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isBoom } from '@hapi/boom';
+import { createValidationFunction } from '../../../common/runtime_types';
+import {
+  inventoryViewRequestParamsRT,
+  INVENTORY_VIEW_URL_ENTITY,
+} from '../../../common/http_api/latest';
+import type { InfraBackendLibs } from '../../lib/infra_types';
+
+export const initDeleteInventoryViewRoute = ({
+  framework,
+  getStartServices,
+}: Pick<InfraBackendLibs, 'framework' | 'getStartServices'>) => {
+  framework.registerRoute(
+    {
+      method: 'delete',
+      path: INVENTORY_VIEW_URL_ENTITY,
+      validate: {
+        params: createValidationFunction(inventoryViewRequestParamsRT),
+      },
+    },
+    async (_requestContext, request, response) => {
+      const { params } = request;
+      const { inventoryViews } = (await getStartServices())[2];
+      const inventoryViewsClient = inventoryViews.getScopedClient(request);
+
+      try {
+        await inventoryViewsClient.delete(params.inventoryViewId);
+
+        return response.noContent();
+      } catch (error) {
+        if (isBoom(error)) {
```

Comment:
Nit: If the id is set to 0 (Default view id) the response is 404, should we have a message explaining that the default view can't be deleted or something like that - because the view is returned from the GET and maybe this can be confusing 🤔

Reply:
In the client will not be possible to delete the static view, and in case they try to do it manually, they'll get the 404 value, but I agree it could be confusing, I'll introduce an error for this case to handle it